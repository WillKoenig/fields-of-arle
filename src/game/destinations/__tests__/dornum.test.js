import dornum, { size } from '../dornum'

const ctx = { currentPlayer: 0 }
const G = {
  players: {
    0: {
      barn: {
        small1: { type: 'plow', contents: [] },
        small2: { type: 'plow', contents: [] },
        small3: { type: 'plow', contents: [] },
        small4: { type: 'plow', contents: [] },
        large1: null,
        large2: null,
        large3: { type: 'plow', contents: [] },
      },
      goods: { food: 0 },
    },
    1: {},
  },
}

describe('dornum load', () => {
  it('has a size', () => {
    expect(size).toEqual(1)
  })

  it('noop', () => {
    expect(dornum([null])({ G, ctx })).toEqual({ G, ctx })
  })

  it('sell only one plow, in large3 first, for 8 food', () => {
    const player = dornum(['plow'])({ G, ctx }).G.players['0']
    expect(player.barn).toMatchSnapshot()
    expect(player.goods.food).toBe(8)
  })
})
