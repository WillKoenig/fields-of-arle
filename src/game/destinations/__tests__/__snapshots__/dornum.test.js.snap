// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dornum load sell one plow, in large3 first, for 8 food 1`] = `
Object {
  "large1": null,
  "large2": null,
  "large3": null,
  "small1": Object {
    "contents": Array [],
    "type": "plow",
  },
  "small2": Object {
    "contents": Array [],
    "type": "plow",
  },
  "small3": Object {
    "contents": Array [],
    "type": "plow",
  },
  "small4": Object {
    "contents": Array [],
    "type": "plow",
  },
}
`;
